### DISCUSSION TITLE
Home Assistant "forgets" all device actions

### DISCUSSION THREAD

**Discussion by LeVraiRoiDHyrule** [Koenkk/zigbee2mqtt#17834] [General]
Hi,

I have 2 Zigbee buttons ([TS0044](https://www.zigbee2mqtt.io/devices/TS0044.html#tuya-ts0044) and https://www.zigbee2mqtt.io/devices/TS0044.html#tuya-ts0044) and they work well with Z2M.

But after a while (around a day I would say), all actions disappear from HA:
![image](https://github.com/Koenkk/zigbee2mqtt/assets/64923542/e66e3864-7a96-444d-8881-95621d0847c8)
![image](https://github.com/Koenkk/zigbee2mqtt/assets/64923542/9d9d49e0-0ef1-4f57-ae94-547177c0c138)

When I do these actions (simple push, double push...) the actions I've done reappear. But the ones I didn't do don't appear. Which means I have to do all actions for them to show in HA, and this every day. It is a big problem because it means I can't create any automation if I didn't do the actions at least a day before. I didn't have this problem with ZHA.

Is that normal ? What could cause this ? What should I do ?
Thanks in advance for any answer, have a great day.

**Comment by saxmanio85** [comment]
I found this discussion searching for the same issue I experience. It happens after a HA or Z2M reboot for me, so I have to push each of my buttons twice before the action is triggered in HA. Not sure if this is a HA or Z2M issue though, anyone have any thoughts?
